src/ChatClient/ChatClientFactory.cs

```diff
@@ -43,14 +43,33 @@ public static IChatClient CreateCopilotChatClientWithGitHubToken()
 
         // Get the Copilot token using the GitHub token
         var helper = new GitHubCopilotHelper();
-        var copilotToken = helper.GetCopilotTokenSync(githubToken);
+        var tokenDetails = helper.GetCopilotTokenDetailsSync(githubToken);
         
-        if (string.IsNullOrEmpty(copilotToken))
+        if (string.IsNullOrEmpty(tokenDetails.token))
         {
             throw new EnvVarSettingException("Failed to get a valid Copilot token from GitHub. Please run 'chatx github login' to authenticate.");
         }
 
-        var options = InitOpenAIClientOptions(endpoint, $"Bearer {copilotToken}");
+        // Create options with the initial auth header
+        var options = InitOpenAIClientOptions(endpoint, $"Bearer {tokenDetails.token}");
+        
+        // Create the refresh policy with a callback that updates both token and expiration
+        CopilotTokenRefreshPolicy refreshPolicy = null!;
+        refreshPolicy = new CopilotTokenRefreshPolicy(
+            tokenDetails.token!, 
```

Comment:
1. Not sure why this assignment isn't on line for declaration... 
2. Since it's "big" (multiline construction, with lambda, refactor somehow to make it more 'one-line' capable)
  * For example, you could move what is the lambda, to a function with the right signature and pass that, or
  * You could take everything to the right of the `=` and make that a function that returns a `new CopilotTokenRefreshPolicy(...)` in this class, or
  * Do ^^^ (right side of `=`) and put it in the GitHubCopilotHelper class as `GetCopilotTokenRefreshPolicy` (or similar)